### PR TITLE
[relay-runtime] augment NormalizationSelection

### PR DESCRIPTION
**Pull request by artola**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/relay-tools/relay-compiler-language-typescript/issues/131#issuecomment-531004495
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/relay-runtime/index.d.ts [hunk 1]
```diff
@@ -728,7 +734,7 @@ interface ReaderCondition {
 }
 
 interface ReaderClientExtension {
-    readonly kind: 'ClientExtension';
+    readonly kind: string; // 'ClientExtension';
```
Comment (alloy) [inline types/relay-runtime/index.d.ts]: Why was this change necessary?

**Review by alloy — CHANGES_REQUESTED** [review]
I think it would make sense to add some test coverage for this, could you add some that shows how you actually use it?

File: types/relay-runtime/index.d.ts (hunk 1 above)
Comment (artola) [reply types/relay-runtime/index.d.ts]: > Why was this change necessary?

Just to match all the other members in the union `ReaderSelection`. I would prefer to have a more strict type than just `string`, but this means that others should be refactored, and possible some other members added to the union that now match `string`.

**Comment by artola** [comment]
> I think it would make sense to add some test coverage for this, could you add some that shows how you actually use it?

Added test for ConcreteRequest.

File: types/relay-runtime/relay-runtime-tests.tsx [hunk 2]
```diff
@@ -75,3 +76,78 @@ commitLocalUpdate(environment, store => {
...
+        },
+        params: {
+            operationKind: 'query',
+            name: 'FooQuery',
+            id: null,
+            text: 'query FooQuery {\n  __typename\n}\n',
+            metadata: {},
+        },
+    };
+})();
```
Comment (alloy) [inline types/relay-runtime/relay-runtime-tests.tsx]: 👌

File: types/relay-runtime/index.d.ts (hunk 1 above)
Comment (alloy) [reply types/relay-runtime/index.d.ts]: Gotcha :+1:

**Comment by alloy** [comment]
The lint failures about the test can just be silenced, but the one in the declaration file should indeed be fixed.

**Comment by artola** [comment]
> The lint failures about the test can just be silenced, but the one in the declaration file should indeed be fixed.

I agree ... working.

**Review by alloy — APPROVED** [review]

**Comment by alloy** [comment]
Perfect, thanks! 🙏
